Branch-TestBed/Branch-TestBed/AppDelegate.m

```diff
@@ -41,15 +41,23 @@ - (BOOL)application:(UIApplication *)application
     // test pre init support
     //[self testDispatchToIsolationQueue:branch]
     
-    
-    [Branch enableLoggingAtLevel:BranchLogLevelVerbose withCallback:^(NSString * _Nonnull message, BranchLogLevel logLevel, NSError * _Nullable error) {
+    [Branch enableLoggingAtLevel:BranchLogLevelVerbose withCallback:^(NSString * _Nonnull message, BranchLogLevel logLevel, NSError * _Nullable error, NSMutableURLRequest * _Nullable request, BNCServerResponse * _Nullable response) {
```

Comment:
so this allows usage of the old function without request and response right?

Reply:
Since they were added to the callback that the function uses itself, there is a slight code change when calling `enableLoggingAtLevel withCallback`

Reply:
ok, does that make this a major change?

Reply:
Since it's just a small callback syntax change, it can still be a minor version bump

Reply:
Does this force developers to change their function call and recompile?

Reply:
@gdeluna-branch Yes, they would have to add `NSMutableURLRequest * _Nullable request, BNCServerResponse * _Nullable response)` to the end of the callback if they're using it, but it wouldn't break or require any changes in their logic.

Reply:
Is there not a way to instead keep the original function signature and have null/empty defaults? Otherwise this is a breaking change.

Reply:
@NidhiDixit09 The change is in BranchLogCallback itself though as opposed to changing enableLogging, so overloading wouldn't work unless we also created another BranchLogCallback.

Overall, this may be a breaking change but I think it's still fine considering the fact that it's a more niche, advanced method/callback and the fix is simple enough to be handled by autocomplete.

Reply:
Yes, we need to define `BranchLogCallbackWithRequestObjects` also in order to overload `enableLogging` method.

Reply:
Yeah I'd rather we extend this until we really want to break it. We/our field teams will get questions if something breaks otherwise.

Reply:
Understandable, but I'm not sure if there's an elegant way to extend it without creating a new function or callback, which may end up being more confusing.
So my vote is just breaking it, but we can try creating a new callback instead if everyone thinks that's better in this case.